lib/matplotlib/tests/test_datetime.py

```diff
@@ -276,11 +276,38 @@ def test_quiverkey(self):
         fig, ax = plt.subplots()
         ax.quiverkey(...)
 
-    @pytest.mark.xfail(reason="Test for scatter not written yet")
     @mpl.style.context("default")
     def test_scatter(self):
-        fig, ax = plt.subplots()
-        ax.scatter(...)
+        mpl.rcParams["date.converter"] = 'concise'
+        base = datetime.datetime(2005, 2, 1)
+        dates = [base + datetime.timedelta(hours=(2 * i)) for i in range(732)]
+        N = len(dates)
+        np.random.seed(19680801)
+        y = np.cumsum(np.random.randn(N))
+        fig, axs = plt.subplots(3, 1, layout='constrained', figsize=(6, 6))
+        lims = [(np.datetime64('2005-02'), np.datetime64('2005-04')),
+                (np.datetime64('2005-02-03'), np.datetime64('2005-02-15')),
+                (np.datetime64('2005-02-03 11:00'), np.datetime64('2005-02-04 13:20'))]
+        axs[0].set_title('Axes.scatter used with datetime on x axis')
```

Comment:
We probably don't need the title here; people won't be looking at the image by itself and can figure this out from the test context.